sklearn/tests/test_pipeline.py

```diff
@@ -125,6 +131,56 @@ def score(self, X, y=None, sample_weight=None):
         return np.sum(X)
 
 
+class DummyTransf(Transf):
+    """Transformer which store the column means"""
+
+    def fit(self, X, y):
+        self.means_ = np.mean(X, axis=0)
+        # Store a timestamp such that we know
+        # that we have a cache object
+        self.timestamp = time.time()
+        return self
+
+
+###############################################################################
+# Test fixtures
+
+env = dict()
+
+
+def setup_module():
+    """ Test setup.
+    """
+    cachedir = mkdtemp()
+    env['dir'] = cachedir
+    if os.path.exists(cachedir):
+        shutil.rmtree(cachedir)
+    # Don't make the cachedir, Memory should be able to do that on the fly
+    print(80 * '_')
+    print('test_memory setup (%s)' % env['dir'])
+    print(80 * '_')
```

Comment:
In fact it was the way which was used in `joblib`.
However, I am having a second thought about that. Probably a simple `try` - `finally` statement as [here](https://github.com/scikit-learn/scikit-learn/blob/4d9a12d175a38f2bcb720389ad2213f71a3d7697/sklearn/cluster/tests/test_hierarchical.py#L132) could be enough for the purpose of the test.